src/coreclr/vm/jitinterface.cpp

```diff
@@ -102,23 +102,35 @@ GARY_IMPL(VMHELPDEF, hlpDynamicFuncTable, DYNAMIC_CORINFO_HELP_COUNT);
 
 #else // DACCESS_COMPILE
 
-uint64_t g_cbILJitted = 0;
-uint32_t g_cMethodsJitted = 0;
+Volatile<int64_t> g_cbILJitted = 0;
+Volatile<int64_t> g_cMethodsJitted = 0;
+Volatile<int64_t> g_c100nsTicksInJit = 0;
+thread_local int64_t g_cbILJittedForThread = 0;
+thread_local int64_t g_cMethodsJittedForThread = 0;
+thread_local int64_t g_c100nsTicksInJitForThread = 0;
 
 #ifndef CROSSGEN_COMPILE
-FCIMPL0(INT64, GetJittedBytes)
+FCIMPL1(INT64, GetCompiledILBytes, CLR_BOOL currentThread)
 {
     FCALL_CONTRACT;
 
-    return g_cbILJitted;
+    return currentThread ? g_cbILJittedForThread : g_cbILJitted.Load();
```

Comment:
Note that volatile Load is not really achieving anything here. It won't fix the value tearing on 32-bit - if it is why you are doing it.

Reply:
It was mainly a non-casting means of disambiguating the ternary expression result. I could swap to a cast or a `LoadWithoutMemoryBarrier` to avoid the extra work of `Load`.

Reply:
My point is that you can drop the `Volatile` on the statics, and it will still work exactly same.

If you wanted to fix the torn read problem, you may want to add a helper method like this https://github.com/dotnet/runtime/pull/55295/files#diff-3d557023dd412cddfc864703ead5c0d6a315721f1207bb9b52849d92d315d7b6R277-R286 that will do a dummy interlocked compared exchange on 32-bit platforms to read the value atomically.

Reply:
That's a good idea. Perhaps outside the scope of this PR, but do we have any need of a reusable form of that helper? I'll do a one-off for this PR, but this seems like a useful snippet to not need to add code for every time.